docs/admin/troubleshooting.md

```diff
@@ -522,14 +522,56 @@ To resolve the issue:
 - Start all `daos_server` processes.
 - Verify that all ranks were able to re-join via `dmg system query`.
 
+### Rebuild is Slow or Running out of Memory
+
+Several environment variables can be set in the env_vars section of the yaml file of the DAOS server
+to control how much memory will be used by the migration service to transfer data while handling
+a rebuild, reintegrate, drain or rebalancing operation of a pool.
+
+The D\_MIGRATE\_ULT\_CNT environment variable controls how many user level threads (i.e. ULT) are
+created at most in the engine while processing one of the operation listed above.
+Each ULT will consume a stack size of 128KiB. To sum up, the following formula defines how much
+memory will be allocated at most for all the ULTs in an engine to manage data migration:
+
+```
+D_MIGRATE_ULT_CNT x 128KiB
+```
+
+The default value for D\_MIGRATE\_ULT\_CNT is 4096. This gives us 0.5GiB of memory used at most
+for the ULT stacks.
+
+Moreover, the D\_MIGRATE\_MEM\_SIZE environment variable defines the maximum total memory in bytes
+allocated by all those migrate ULTs to buffer the data.
```

Comment:
not very sure if this description is accurate, if user think that rebuild is limited to use these amount of memory to rebuild a pool, then that possibly is not accurate. For example rebuild scan possibly can use more memory for enumerate.

Reply:
If the enumerate means migrate_one_epoch_object() -> dsc_obj_list_obj(), looks the memory used there is not controlled by tls->mpt_inflight_size/mpt_inflight_max_size.

Reply:
Right, but it is allocating everything on the stack (which is pretty bad in itself and should eventually be fixed), but not allocating extra memory.